app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/LoginScreenViewModel.kt

```diff
@@ -23,16 +31,30 @@ class LoginScreenViewModel @Inject constructor(private val spotifyController: Sp
   fun handleAuthorizationResponse(response: AuthorizationResponse) {
     when (response.type) {
       AuthorizationResponse.Type.TOKEN -> {
-        _uiState.value = UiState(hasResult = true, success = true)
+        authenticate(response.accessToken)
       }
       AuthorizationResponse.Type.ERROR -> {
         _uiState.value =
-            UiState(
+            uiState.value.copy(
                 hasResult = true, success = false, message = "Error logging in: ${response.error}")
       }
       else -> {
         _uiState.value =
-            UiState(hasResult = true, success = false, message = "User cancelled login")
+            uiState.value.copy(hasResult = true, success = false, message = "User cancelled login")
+      }
+    }
+  }
+
+  private fun authenticate(token: String) {
+    CoroutineScope(Dispatchers.IO).launch {
```

Comment:
Why coroutine scope and not viewmodelscope ?

Reply:
I think I once had issues with that because the viewModelScope would stil run on the UI thread, which would not work in some cases. Using the IO scope has worked better so far